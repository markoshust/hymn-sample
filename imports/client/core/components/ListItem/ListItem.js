import React, { PropTypes } from 'react';
import { compose, withHandlers } from 'recompose';

const ListItem = ({
  children,
  isStriked,
  onClick,
}) => (
  <li
    onClick={onClick}
    style={{ textDecoration: isStriked ? 'line-through' : 'none' }}
  >
    {children}
  </li>
);

ListItem.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string,
  isStriked: PropTypes.bool,
  onClick: PropTypes.func,
};

const enhance = compose(
  withHandlers({
    onClick: ({
      id,
      onClick,
    }) => () => onClick && onClick(id),
  })
);

export default enhance(ListItem);
