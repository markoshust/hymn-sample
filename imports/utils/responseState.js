const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
