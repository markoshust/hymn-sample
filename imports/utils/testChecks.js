import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
