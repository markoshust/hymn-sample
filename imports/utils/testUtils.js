import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
