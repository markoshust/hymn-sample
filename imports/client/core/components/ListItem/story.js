import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { boolean, text } from '@kadira/storybook-addon-knobs';
import ListItem from './ListItem';

const stories = storiesOf('core/ListItem', module);

stories.add('Standard', () => (
  <ListItem onClick={action('click')}>
    {text('Text', 'Some Text')}
  </ListItem>
));

stories.add('Striked', () => (
  <ListItem
    isStriked={boolean('Is Striked', true)}
    onClick={action('click')}
  >
    {text('Text', 'Some Text')}
  </ListItem>
));
