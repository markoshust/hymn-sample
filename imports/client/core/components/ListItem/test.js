import React from 'react';
import { renderCheck } from 'utils/testChecks';
import ListItem from './ListItem';

const onClick = jest.fn();

describe('<ListItem />', () => {
  it('renders with text', () => {
    renderCheck(
      <ListItem>Some Text</ListItem>
    );
  });

  it('renders with text and onClick', () => {
    renderCheck(
      <ListItem onClick={onClick}>Some Text</ListItem>
    );
  });

  it('renders with isStriked', () => {
    renderCheck(
      <ListItem isStriked>Striked Out</ListItem>
    );
  });
});
