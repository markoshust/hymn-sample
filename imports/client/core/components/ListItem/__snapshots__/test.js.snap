exports[`<ListItem /> renders with isStriked 1`] = `
<li
  onClick={[Function anonymous]}
  style={
    Object {
      "textDecoration": "line-through"
    }
  }>
  Striked Out
</li>
`;

exports[`<ListItem /> renders with text 1`] = `
<li
  onClick={[Function anonymous]}
  style={
    Object {
      "textDecoration": "none"
    }
  }>
  Some Text
</li>
`;

exports[`<ListItem /> renders with text and onClick 1`] = `
<li
  onClick={[Function anonymous]}
  style={
    Object {
      "textDecoration": "none"
    }
  }>
  Some Text
</li>
`;
